### PR TITLE
Refactor BaseGraphQlTest into BaseAuthenticatedFullStackTest

### LINKED ISSUE #4848: Refactor BaseGraphQlTest into BaseAuthenticatedFullStackTest
### Background
The patient bulk upload async service is using BaseGraphqlTest as its base, because that class offers all the benefits of BaseFullStackTest plus some additional authentication setup. PatientBulkUploadServiceAsyncTest is not however related to GraphQL at all, so it probably should extend something else.

### Action requested
Refactor BaseGraphQlTest into two separate base tests - `BaseAuthenticatedFullStackTest` and `BaseGraphQlTest`.

`BaseAuthenticatedFullStackTest` should extend `BaseFullStackTest` and include some of the setup/teardown that `BaseGraphQlTest` currently offers, as well as all the authentication-specific methods therein.

`BaseGraphQlTest` should extend `BaseAuthenticatedFullStackTest` and include any graphql-specific methods, like the various `runQuery` methods.

`PatientBulkUploadServiceAsyncTest` should also be changed to extend `BaseAuthenticatedFullStackTest` instead of `BaseGraphQlTest`.

### Acceptance Criteria
1. `PatientBulkUploadServiceAsyncTest` no longer extends `BaseGraphQlTest`.
2. All the other tests that extend `BaseGraphQlTest` still pass.

### Additional context
See discussion in PR #4722: https://github.com/CDCgov/prime-simplereport/pull/4722#discussion_r1046391602

### PR DESCRIPTION
**Pull request by emmastephenson**
# BACKEND PULL REQUEST

## Related Issue

- Fixes #4848 

## Changes Proposed

- BaseGraphQlTest is split into two files - BaseGraphQlTest and BaseAuthenticatedFullStackTest
- PatientBulkUploadServiceAsyncTest extends BaseAuthenticatedFullStackTest

## Additional Information

- Had to rename the `@BeforeEach` and `@AfterEach` methods in BaseAuthenticatedFullStackTest to avoid overwriting them in BaseGraphQlTest

## Testing

- Tests pass

### CODE REVIEW

File: backend/src/test/java/gov/cdc/usds/simplereport/api/BaseAuthenticatedFullStackTest.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+  @Autowired private OrganizationInitializingService organizationInitializingService;
+  @Autowired private DemoOktaRepository oktaRepository;
+  @Autowired private DemoUserConfiguration userConfiguration;
+  @MockBean private AddressValidationService addressValidationService;
+
+  @Getter @Setter private String username = null;
+
+  protected void useOrgUser() {
+    username = TestUserIdentities.STANDARD_USER;
```
Comment (emyl3) [inline backend/src/test/java/gov/cdc/usds/simplereport/api/BaseAuthenticatedFullStackTest.java:36]: just to make sure I'm understanding correctly, removing the underscore  throughout was to adhere to the style guide right? https://github.com/CDCgov/prime-simplereport/wiki/SimpleReport-Style-Guide#variable-names 🧐

Comment (emmastephenson) [reply backend/src/test/java/gov/cdc/usds/simplereport/api/BaseAuthenticatedFullStackTest.java:36]: Yep! Exactly. Might as well fix the naming while touching the file anyways :)

Comment (emyl3) [reply backend/src/test/java/gov/cdc/usds/simplereport/api/BaseAuthenticatedFullStackTest.java:36]: I see! Thank you!! 🎉

**Review by BobanL — APPROVED** [review]
Looks good to me!

**Review by emyl3 — APPROVED** [review]
LGTM! All the BaseGraphQlTests passed for me locally as well! Thank you, Emma! 🧹

**Pull request merged**